site/cds_rdm/inspire_harvester/transform_entry.py

```diff
@@ -146,69 +184,251 @@ def _transform_publication_date(self):
+        # format: "© {holder} {year}, {statement} {url}"
+        copyrights = self.inspire_metadata.get("copyright", [])
+        result_list = []
+
+        for cp in copyrights:
+            holder = cp.get("holder", "")
+            statement = cp.get("statement", "")
+            url = cp.get("url", "")
+            year = str(cp.get("year", ""))
+
+            if not any([holder, statement, url, year]):
+                return None
+            else:
+                parts = []
+                if holder or year:
+                    holder_year = " ".join(filter(None, [holder, year]))
+                    parts.append(f"{holder_year}")
+                if statement or url:
+                    statement_url = " ".join(filter(None, [statement, url]))
+                    parts.append(statement_url)
+                rdm_copyright = "© " + ", ".join(parts)
+
+                result_list.append(rdm_copyright)
+        return "<br />".join(result_list)
+
+    def _transform_dois(self):
+        """Mapping of record dois."""
+        DATACITE_PREFIX = current_app.config["DATACITE_PREFIX"]
+        dois = self.inspire_metadata.get("dois", [])
+
+        if len(dois) > 1:
+            self.metadata_errors.append(
+                f"More than 1 DOI was found in the INSPIRE record #{self.inspire_metadata.get('control_number')}."
+            )
+            return None
+        elif len(dois) == 0:
+            return None
+        else:
+            doi = dois[0].get("value")
+            if is_doi(doi):
+                mapped_doi = {
+                    "identifier": doi,
+                }
+                if doi.startswith(DATACITE_PREFIX):
+                    mapped_doi["provider"] = "datacite"
+                else:
+                    mapped_doi["provider"] = "external"
+                return mapped_doi
+            else:
+                self.metadata_errors.append(
+                    f"DOI validation failed. Value: {doi}. INSPIRE record #{self.inspire_metadata.get('control_number')}."
+                )
+                return None
+
     def _transform_alternate_identifiers(self):
         """Mapping of alternate identifiers."""
         identifiers = []
         inspire_id = self.inspire_metadata.get("control_number")
+        RDM_RECORDS_IDENTIFIERS_SCHEMES = current_app.config[
+            "RDM_RECORDS_IDENTIFIERS_SCHEMES"
+        ]
+        IDENTIFIERS_SCHEMES_TO_DROP = current_app.config["IDENTIFIERS_SCHEMES_TO_DROP"]
+
         try:
             # add INSPIRE id
             identifiers.append({"identifier": str(inspire_id), "scheme": "inspire"})
+
+            # external_system_identifiers
+            external_sys_ids = self.inspire_metadata.get(
+                "external_system_identifiers", []
+            )
+            for external_sys_id in external_sys_ids:
+                schema = external_sys_id.get("schema").lower()
+                if schema == "cds":
+                    schema = "lcds"
```

Comment:
we will need to align with Inspire on how to treat new CDS pids, can you add a task to track this?